src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/search/parameter/SearchRequestCustomDeserializer.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.search.parameter;
+
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import org.springframework.boot.jackson.JsonComponent;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.search.SearchOperator;
+import uk.gov.hmcts.reform.wataskmanagementapi.exceptions.BadRequestException;
+
+import java.io.IOException;
+
+@JsonComponent
+@SuppressWarnings({"PMD.LawOfDemeter"})
+public class SearchRequestCustomDeserializer extends StdDeserializer<SearchParameter<?>> {
+
+    private static final long serialVersionUID = -1895766495984179418L;
+
+    private static final String ERROR_MESSAGE =
+        "Each search_parameter element must have 'key', 'values' and 'operator' fields present and populated.";
+
+    public SearchRequestCustomDeserializer() {
+        this(null);
+    }
+
+    public SearchRequestCustomDeserializer(final Class<?> cls) {
+        super(cls);
+    }
+
+    @Override
+    public SearchParameter<?> deserialize(JsonParser jsonParser, DeserializationContext ctxt) throws IOException {
+
+        final ObjectMapper mapper = (ObjectMapper) jsonParser.getCodec();
+        final JsonNode searchNode = mapper.readTree(jsonParser);
+
+        final JsonNode operatorNode = searchNode.get("operator");
+
+        if (operatorNode == null) {
+            throw new BadRequestException(ERROR_MESSAGE);
+        }
+
+        if (SearchOperator.BOOLEAN.getValue().equals(operatorNode.asText())) {
+            return mapper.treeToValue(searchNode, SearchParameterBoolean.class);
+        } else if (SearchOperator.IN.getValue().equals(operatorNode.asText())
+                   || SearchOperator.AFTER.getValue().equals(operatorNode.asText())) {
```

Comment:
ExUI have completed the change and update their erroneous consumer "AFTER" https://tools.hmcts.net/jira/browse/EUI-4940. We can remove "AFTER".